### DISCUSSION TITLE
The first query of data after initialization needs to be delayed？

### DISCUSSION THREAD

**Discussion by wang12190529** [ujjwalguptaofficial/JsStore#369] [Q&A]
Whenever I reopen my web site, await connection.initDb(dataBase); is called to initialize or reload my database, but every time I pass connection.select for the first time, I cannot query the correct results, it It always returns empty, but I delay the query for 3-5 seconds and it works fine, can anyone provide me with any help?
jsstore version :4.7.2

**Comment by ujjwalguptaofficial** [comment]
Could you please turn on your log and share your screenshot.

https://jsstore.net/docs/helpers#logstatus

**Comment by wang12190529** [comment]
![Chrome console](https://github.com/user-attachments/assets/dcee9e42-4347-49fd-bc24-a35b444ec805)
```
async loadGalleryAll(options: {
        cameraCondition?: IWhereQuery,
        typeCondition?: IWhereQuery,
        dateCondition?: IWhereQuery
    }) {
        return new Promise<GalleryVo[]>((resolve, reject) => {
            console.log('loadGalleryAll options', options);
            const query: ISelectQuery = {
                from: DB_TABLE_NAME.GALLERY,
                order: [{
                    by: 'folder',
                    type: 'desc'
                }, {
                    by: 'mediaId',
                    type: 'desc'
                },]
            };
            const whereArr: any[] = [{
                simDeviceId: {
                    in: this.simDeviceIds
                },
                lastModified: {
                    '>': this.MAX_DAY
                }
            }];

            if (options.cameraCondition != undefined) {
                console.log('loadGalleryAll options.cameraCondition', options.cameraCondition);
                whereArr.push(options.cameraCondition);
            } else {
                console.log('loadGalleryAll cameraCondition', whereArr);
                whereArr.push({
                    simDeviceId: {
                        in: this.simDeviceIds
                    }
                });
            }
            if (options.typeCondition != undefined) {
                console.log('loadGalleryAll typeCondition', whereArr);
                whereArr.push(options.typeCondition);
            }
            if (options.dateCondition != undefined) {
                whereArr.push(options.dateCondition);
            }
            query.where = whereArr;
            console.log('loadGalleryAll query', query, whereArr);
            connection.select(query).then((res: any[]) => {
                console.log('loadGalleryAll res', res);
                resolve(res);
                if (res.length > 0) {
                    this.rePresignedUrlsQueue();
                    this.syncMetaQueue();
                }
            }).catch(err => {
                reject(err);
            });
        });
    };
```

**Reply by wang12190529** [reply]
The first query returns [], and the second query returns the correct data list. All conditions are the same.

**Reply by wang12190529** [reply]
Add delay and it works fine,I don't understand why
```
    delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
    async loadGalleryAll(options: {
        cameraCondition?: IWhereQuery,
        typeCondition?: IWhereQuery,
        dateCondition?: IWhereQuery
    }) {
        await this.delay(5000);
        return new Promise<GalleryVo[]>((resolve, reject) => {
    .......
```

**Comment by wang12190529** [comment]
![logStatus1](https://github.com/user-attachments/assets/5d15cd64-8ccb-4ab1-a2b0-2cea08cb29c3)
![logStatus2](https://github.com/user-attachments/assets/d7ac0845-eeab-4a55-ba60-4039188c4023)

**Reply by wang12190529** [reply]
`request pushed: map_set`
 The query after the log is normal.

**Comment by wang12190529** [comment]
![1](https://github.com/user-attachments/assets/0cd6dd4a-4120-40b4-9b60-9d60ba86c9d3)
![2](https://github.com/user-attachments/assets/f7936a9f-6965-4e31-a038-13366218098f)
Comparison of delayed and non-delayed logs, as well as map_set log printing timing

**Comment by ujjwalguptaofficial** [comment]
from log, everything looks good to me. I think there might be problem with the query, may be when you are calling after delay - query is different. Could you log the query before calling the select api and compare between delay and non delay.

**Reply by wang12190529** [reply]
The query is completely the same, the same query function, the same call. I only need to add a delay before querying the api call to get a normal return. Otherwise, it will return empty. This happens every time I refresh the page initDb, and the screenshot has an increased delay. Code examples with and without deferral.

**Reply by wang12190529** [reply]
![image](https://github.com/user-attachments/assets/763e0cd3-6f9e-4145-929c-e0555575f193)

Now I can only rely on increasing the delay by 5000ms to ensure that the first query is normal.

**Comment by ujjwalguptaofficial** [comment]
Could you please try updating to the latest version of jsstore ?

**Reply by wang12190529** [reply]
Okay, I will give you feedback after trying it

**Reply by wang12190529** [reply]
upgrade to "jsstore": "^4.8.1"  
But still the same result

**Reply by ujjwalguptaofficial** [reply]
could you please provide me anything reproducible - a small POC would be great.
